subsys/bluetooth/mesh/shell/shell_utils.c

```diff
@@ -48,6 +51,74 @@ bool shell_model_str2bool(const char *str)
 	return str2u8(str);
 }
 
+
+int shell_model_str2sensorval(const char *str, struct sensor_value *out)
+{
+	int32_t sign = 1;
+	char buf[18]; /* 10 digits + decimal point + 6 digits + terminator */
+
+	if (str != NULL && *str == '-') {
+		sign = -1;
+		str++;
+	}
+
+	if (strlen(str) > 17 || strlen(str) == 0) {
+		return -EINVAL;
+	}
+
+	strcpy(buf, str);
+
+	char *dp = strchr(buf, '.');
+
+	if (dp != NULL) {
+		*dp = 0;
+		dp++;
+	}
+
+	errno = 0;
+	uint32_t val = strtoul(buf, NULL, 10);
+
+	if (errno || (sign == 1 && val > INT32_MAX) || (sign == -1 && val > INT32_MAX + 1ul)) {
+		return -EINVAL;
+	}
+	out->val1 = sign * val;
```

Comment:
Won't `sign` be promoted to `uint32_t` here? Probably, `val` should be casted to `int32_t` here. Same for `val2`.